node/src/types/value_or_chunk.rs

```diff
@@ -0,0 +1,164 @@
+use std::fmt::{self, Debug, Display, Formatter};
+
+use casper_execution_engine::storage::trie::TrieRaw;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use casper_hashing::{ChunkWithProof, Digest, MerkleConstructionError};
+use datasize::DataSize;
+
+/// Represents a value or a chunk of data with attached proof.
+/// Chunk with attached proof is used when the requested
+/// value is larger than [ChunkWithProof::CHUNK_SIZE_BYTES].
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum ValueOrChunk<V> {
+    /// Represents a value.
+    Value(V),
+    /// Represents a chunk of data with attached proof.
+    ChunkWithProof(ChunkWithProof),
+}
+
+#[derive(Debug, Error)]
+/// Error returned when constructing an instance of `ValueOrChunk`.
+pub enum ChunkingError {
+    /// Merkle proof construction error
+    #[error("error constructing merkle proof for chunk")]
+    MerkleConstruction(#[from] MerkleConstructionError),
+    /// Serialization error
+    #[error("error serializing data: {0}")]
+    SerializationError(#[source] bincode::Error),
+}
+
+impl<V> ValueOrChunk<V> {
+    /// Creates an instance of `ValueOrChunk::Value` if data is less than `max_chunk_size`
```

Comment:
```suggestion
    /// Creates an instance of `ValueOrChunk::Value` if data size is less than or equal to [ChunkWithProof::CHUNK_SIZE_BYTES]
```

Reply:
Done in [`e8243e8` (#3255)](https://github.com/casper-network/casper-node/pull/3255/commits/e8243e8b08580a6e72cebdfed0b1b1957f46652a)